src/System.Private.CoreLib/shared/System/Threading/ReaderWriterLockSlim.cs

```diff
@@ -1,1681 +1,1681 @@
-                    // thread to stop spinning and hopefully enter a proper wait state.
-                    Debug.Assert(
-                        reason == EnterSpinLockReason.EnterAnyRead ||
-                        reason == EnterSpinLockReason.EnterWrite ||
-                        reason == EnterSpinLockReason.UpgradeToWrite);
-                    if (spinIndex >= (LockSpinCount + LockSleep0Count + DeprioritizedLockSleep1Count))
-                    {
-                        reason |= EnterSpinLockReason.Wait;
-                        spinIndex = -1;
-                    }
-                }
-            }
-
-            public void Exit()
-            {
-                Debug.Assert(_isLocked != 0, "Exiting spin lock that is not held");
-                Volatile.Write(ref _isLocked, 0);
-            }
-
-#if DEBUG
-            public bool IsHeld => _isLocked != 0;
-#endif
-        }
-
-        [Flags]
-        private enum WaiterStates : byte
-        {
-            None = 0x0,
-
-            // Used for quick check when there are no waiters
-            NoWaiters = 0x1,
-
-            // Used to avoid signaling more than one waiter to wake up when only one can make progress, see WaitOnEvent
-            WriteWaiterSignaled = 0x2,
-            UpgradeableReadWaiterSignaled = 0x4
-            // Write upgrade waiters are excluded because there can only be one at any given time
-        }
-
-        private enum EnterSpinLockReason
-        {
-            EnterAnyRead = 0,
-            ExitAnyRead = 1,
-            EnterWrite = 2,
-            UpgradeToWrite = 3,
-            EnterRecursiveWrite = 4,
-            ExitAnyWrite = 5,
-
-            OperationMask = 0x7,
-
-            Wait = 0x8
-        }
-
-        private enum EnterLockType
-        {
-            Read,
-            UpgradeableRead,
-            Write,
-            UpgradeToWrite
-        }
-    }
-}
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Diagnostics; // for TraceInformation
+using System.Runtime.CompilerServices;
+
+namespace System.Threading
+{
+    public enum LockRecursionPolicy
+    {
+        NoRecursion = 0,
+        SupportsRecursion = 1,
+    }
+
+    //
+    // ReaderWriterCount tracks how many of each kind of lock is held by each thread.
+    // We keep a linked list for each thread, attached to a ThreadStatic field.
+    // These are reused wherever possible, so that a given thread will only
+    // allocate N of these, where N is the maximum number of locks held simultaneously
+    // by that thread.
+    // 
+    internal class ReaderWriterCount
+    {
+        // Which lock does this object belong to?  This is a numeric ID for two reasons:
+        // 1) We don't want this field to keep the
```

Comment:
did you miss the field? `private static ReaderWriterCount t_rwc;` -> `private static ReaderWriterCount? t_rwc;`

Reply:
> did you miss the field?

In general I've not been proactively annotating all statics, since the compiler doesn't warn about them.  I expect there are many such statics across the files we've annotated that we haven't touched.  We should do a follow-up pass to clean those up.  And then hopefully https://github.com/dotnet/roslyn/issues/34668 will be addressed and the compiler will help flag them for us.